### PR TITLE
RBD action size validation

### PR DESCRIPTION
**Pull request by wwwtyro**
@mbruzek @chuckbutler @Cynerva

### CODE REVIEW

**Comment by Cynerva** [comment]
Built locally and tested.

Example failure case:
```
$ juju run-action kubernetes-master/0 create-rbd-pv name=test size=8103
Action queued with id: eacde62d-af1f-404f-8d97-b4f623d014c9
$ juju show-action-output eacde62d-af1f-404f-8d97-b4f623d014c9
message: Unable to allocate RBD of size 8103MB, only 8102MB are available.
status: failed
timing:
  completed: 2016-10-19 20:54:08 +0000 UTC
  enqueued: 2016-10-19 20:54:06 +0000 UTC
  started: 2016-10-19 20:54:07 +0000 UTC
```

Example success:
```
$ juju run-action kubernetes-master/0 create-rbd-pv name=test size=8102
Action queued with id: bd5096d2-9797-4ebc-8999-8cced09863a6
$ juju show-action-output bd5096d2-9797-4ebc-8999-8cced09863a6
status: completed
timing:
  completed: 2016-10-19 20:54:36 +0000 UTC
  enqueued: 2016-10-19 20:54:29 +0000 UTC
  started: 2016-10-19 20:54:32 +0000 UTC
```

Also verified that the PV looks correct in kubectl.

Example of `skip-size-check` allowing us to bypass the check:
```
$ juju run-action kubernetes-master/0 create-rbd-pv name=test3 size=20000 skip-size-check=true
Action queued with id: ff098778-5a52-4277-8f7b-2a848a557161
$ juju show-action-output ff098778-5a52-4277-8f7b-2a848a557161
status: completed
timing:
  completed: 2016-10-19 20:56:10 +0000 UTC
  enqueued: 2016-10-19 20:56:02 +0000 UTC
  started: 2016-10-19 20:56:07 +0000 UTC
```

Looks good to me, but I'd like @chuckbutler or @mbruzek to weigh in on code review also

File: cluster/juju/layers/kubernetes-master/actions/create-rbd-pv [hunk 1]
```diff
@@ -153,6 +159,20 @@ def get_monitors():
...
+    the RBD pool ('rbd') isn't found. '''
+    command = ['ceph', 'df']
+    debug_command(command)
+    out = check_output(command).decode('utf-8')
+    for line in out.splitlines():
+        stripped = line.strip()
+        if stripped.startswith('rbd'):
+            M = stripped.split()[-2].replace('M', '')
+            return int(M)
+    raise UnknownAvailableSpaceException('Unable to determine available space.')  # noqa
```
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-master/actions/create-rbd-pv]: That is a long tail exception right there gentlemen. 🤓

File: cluster/juju/layers/kubernetes-master/actions/create-rbd-pv [hunk 2]
```diff
@@ -203,6 +223,17 @@ def validate_relation():
     return True
 
 
+def validate_space(size):
+    if action_get_or_default('skip-size-check'):
+        return True
+    available_space = get_available_space()
+    if available_space < size:
```
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-master/actions/create-rbd-pv]: nice, almost no math required.

File: cluster/juju/layers/kubernetes-master/actions/create-rbd-pv [hunk 3]
```diff
@@ -203,6 +223,17 @@ def validate_relation():
...
+        msg = 'Unable to allocate RBD of size {}MB, only {}MB are available.'
+        action_fail(msg.format(size, available_space))
```
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-master/actions/create-rbd-pv]: Seems weird to fail the action this deep in the call stack... but I see you're using the space values in the messaging here. I'm +1 to this

**Comment by mbruzek** [comment]
Yeah this looks like good RDB stuff! +1 from me!

**Pull request merged**